### ISSUE TITLE
Ingame Help Menu not loading completely

### ISSUE THREAD

**Issue by DreiFingerJoe** [bodzio528/FS22_CropRotation#37]
Hello, when crop rotation is in mod folder any ingame help menu after cr in alphabet will not be loaded.

**Comment by Gordon-Dry** [comment]
Oh, that is interesting. Testing this by changing the mod folder's name to FS22_zz4_CropRotation (I already got zz1 to zz3 over there) to check the issue with another mod's help menu not appearing.

**Comment by Gordon-Dry** [comment]
Yes, ^that fixes the issue.

**Comment by DreiFingerJoe** [comment]
Hi, thank you for that idea, it is working for all ingame help, except precision farming. Maybe PF has an error in it too. With my little knowledge about programming, I can see PF is the only mod loading helpLine.xml from its xml, all outher mods load from lua.

**Comment by DreiFingerJoe** [comment]
CP had this error last year. After comparing the old with the new version, I have fond a difference in how help menu was loaded and added this part to croprotation.lua. Now ingame help is working fine.

function loadCropRotationHelpLine(superFunc, ...)
    local ret = superFunc(self, ...)
	g_cropRotation.log:debug("loadCropRotationHelpLine()")
	if ret then

    self:loadFromXML(Utils.getFilename("gui/helpLine.xml", g_cropRotation.MOD_DIRECTORY))
    return true
	return false
end

**Comment by Gordon-Dry** [comment]
You closed this issue because you found a private solution?
This issue should stay open until the mod is officially fixed, this is how Github works.
You even should provide a pull request with the fix.

**Comment by Gordon-Dry** [comment]
@DreiFingerJoe 
I guess your block should be
```
function loadCropRotationHelpLine(superFunc, ...)
	local ret = superFunc(self, ...)
	g_cropRotation.log:debug("loadCropRotationHelpLine()")
	if ret then
		self:loadFromXML(Utils.getFilename("gui/helpLine.xml", g_cropRotation.MOD_DIRECTORY))
		return true
	else
		return false
	end
end
```

**Comment by DreiFingerJoe** [comment]
As I am new to GitHub, could you please assist me with creating a proper pull request?

**Comment by Gordon-Dry** [comment]
Why me? I learned it for myself, I asked nobody. Read the docs:
https://docs.github.com/en/pull-requests

**Comment by Gordon-Dry** [comment]
^btw the above fix may work for CP but not for the other mods mentioned above, so I have to go back to the solution by renaming the mod zip file.

**Comment by Gordon-Dry** [comment]
@DreiFingerJoe btw changing that function to a _superFunc_ makes it throw an error:
```
2023-11-02 19:51 Error: Running LUA method 'update'.
G:/Admin/Documents/My Games/FarmingSimulator2022/mods/FS22_zz4_CropRotation/CropRotation.lua:661: attempt to call local 'superFunc' (a table value)

```

**Comment by DreiFingerJoe** [comment]
This is the code I am using and I have no errors.


`function loadCropRotationHelpLine(superFunc, ...)
	local ret = superFunc(self, ...)
		g_cropRotation.log:debug("loadCropRotationHelpLine()")
		if ret then
		self:loadFromXML(Utils.getFilename("gui/helpLine.xml", g_cropRotation.MOD_DIRECTORY))
		return true
	return false
end
HelpLineManager.loadMapData = Utils.overwrittenFunction(HelpLineManager.loadMapData, loadCropRotationHelpLine)`

**Comment by DreiFingerJoe** [comment]
This is the code I am using and it works without errors

```
function loadCropRotationHelpLine(superFunc, ...)
	local ret = superFunc(self, ...)
		g_cropRotation.log:debug("loadCropRotationHelpLine()")
		if ret then
		self:loadFromXML(Utils.getFilename("gui/helpLine.xml", g_cropRotation.MOD_DIRECTORY))
		return true
	return false
end
HelpLineManager.loadMapData = Utils.overwrittenFunction(HelpLineManager.loadMapData, loadCropRotationHelpLine)
```

**Comment by Gordon-Dry** [comment]
Well, the line
`local ret = superFunc(self, ...)`
triggers the error on my side.

It is what it is.

**Comment by lazarov27** [comment]
I had to make similar correction because I found out Crop Rotation was breaking Precision Farming's Help System (it disappears, and clicking on Help button in PF screen shows Crop Rotation help page).
Here's what I'm using with no issues so far (borrowed from Courseplay's implemenation)
```lua
function CropRotation:loadMapDataHelpLineManager(superFunc, ...)
	local ret = superFunc(self, ...)
	if ret then
		self:loadFromXML(Utils.getFilename("gui/helpLine.xml", g_cropRotation.MOD_DIRECTORY))
		return true
	end
	return false
end
HelpLineManager.loadMapData = Utils.overwrittenFunction( HelpLineManager.loadMapData, CropRotation.loadMapDataHelpLineManager)
```

**Comment by Gordon-Dry** [comment]
I can confirm that this one works.

**Comment by bodzio528** [comment]
Thank you for report and input. Indeed, crop rotation helpline loading function have had a bug that caused all subsequent help menu loading ineffective. 

Should be fixed in 2.5.0.3

**Comment by FirenzeIT** [comment]
[image: d22eff6a371263cb88ddaf550a52683a644f0cec]​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​​Hello,
i opened another issue with v2.5.0.3 stuck load at 95%,but with 2.3.0.1 not
having any issues (but it's missing a new culture obv), so if possible,
it's need work to 2.3.0.1 and add here the new colture, my personal
opinion)...

Il Gio 7 Dic 2023, 23:54 Bogumił Chojnowski ***@***.***> ha
scritto:
